servers/rendering/renderer_rd/shaders/scene_forward.glsl

```diff
@@ -591,14 +598,25 @@ LIGHT_SHADER_CODE
 #endif
 
 #if defined(SPECULAR_BLINN) || defined(SPECULAR_SCHLICK_GGX) || defined(LIGHT_CLEARCOAT_USED)
+#ifdef USE_SOFT_SHADOWS
 	float cNdotH = clamp(A + dot(N, H), 0.0, 1.0);
+#else
+	float cNdotH = clamp(dot(N, H), 0.0, 1.0);
+#endif
 #endif
 
 #if defined(DIFFUSE_BURLEY) || defined(SPECULAR_SCHLICK_GGX) || defined(LIGHT_CLEARCOAT_USED)
+#ifdef USE_SOFT_SHADOWS
 	float cLdotH = clamp(A + dot(L, H), 0.0, 1.0);
+#else
+	float cLdotH = clamp(dot(L, H), 0.0, 1.0);
+#endif
 #endif
 
+	float metallic = unpackUnorm4x8(orms).z;
 	if (metallic < 1.0) {
+		float roughness = unpackUnorm4x8(orms).y;
+
 #if defined(DIFFUSE_OREN_NAYAR)
```

Comment:
If deleting Oren-Nayar, it should be deleted here and in the material and shader as well

Reply:
Fixed by https://github.com/godotengine/godot/pull/49419